Fix check-deploy skipping the root path

SonicHedgehog opened the pull request:
I noticed that `check-deploy` was no longer correctly checking the root path, `/`, due to [6eefcbb](https://github.com/progrium/dokku/commit/6eefcbb87c1d13d3ff0623a7fd58d37261112d71). This commit fixes this by not requiring any characters after the slash.

michaelshobbs commented:
A rebase from master should help this failed test now as well.

josegonzalez commented:
Looks good to me.

josegonzalez commented:
Could use a test maybe though?

michaelshobbs commented:
@SonicHedgehog maybe just change one of the test app's `CHECKS` files to test this format?

SonicHedgehog commented:
I have now added a `checks-root` test app which contains a check that should fail. To make testing this app possible, I have added a parameter to the `test_deploy` script to allow for apps that *should* intentionally fail deployment.

@michaelshobbs Almost all test apps (for example, the [`go` test app](https://github.com/progrium/dokku/blob/master/tests/apps/go/CHECKS)) already have a check in this format but it didn’t test if the check was actually executed.

michaelshobbs commented:
@SonicHedgehog I'm not following. I see when I deploy the go app that the check is being evaluated. No?
```
...
remote: + pluginhook check-deploy test-go-3397 113fa0b2aea064b99b98fc7395f0d403f506ae6f2542cc99dc939f8e10375a19 web 5000 172.17.0.19
-----> Running pre-flight checks
remote: ++ dirname /var/lib/dokku/plugins/20_events/check-deploy
remote: + source /var/lib/dokku/plugins/20_events/../common/functions
remote: ++ set -eo pipefail
remote: ++ [[ -n 1 ]]
remote: ++ set -x
remote: + [[ ! -n '' ]]
remote: ++ dirname /var/lib/dokku/plugins/checks/check-deploy
remote: + source /var/lib/dokku/plugins/checks/../common/functions
remote: ++ set -eo pipefail
remote: ++ [[ -n 1 ]]
remote: ++ set -x
remote: + APP=test-go-3397
remote: + DOKKU_APP_CONTAINER_ID=113fa0b2aea064b99b98fc7395f0d403f506ae6f2542cc99dc939f8e10375a19
remote: + DOKKU_APP_CONTAINER_TYPE=web
remote: + DOKKU_APP_LISTEN_PORT=5000
remote: + DOKKU_APP_LISTEN_IP=172.17.0.19
remote: + [[ -z 5000 ]]
remote: + [[ -z 172.17.0.19 ]]
remote: + [[ -z 113fa0b2aea064b99b98fc7395f0d403f506ae6f2542cc99dc939f8e10375a19 ]]
remote: + [[ -f /home/dokku/ENV ]]
remote: + source /home/dokku/ENV
remote: ++ export CURL_CONNECT_TIMEOUT=5
remote: ++ CURL_CONNECT_TIMEOUT=5
remote: ++ export CURL_TIMEOUT=30
remote: ++ CURL_TIMEOUT=30
remote: + [[ -f /home/dokku/test-go-3397/ENV ]]
remote: + source /home/dokku/test-go-3397/ENV
remote: + WAIT=5
remote: + TIMEOUT=30
remote: + ATTEMPTS=5
remote: ++ mktemp -d /tmp/CHECKS.XXXXX
remote: + TMPDIR=/tmp/CHECKS.Xg2D4
remote: + docker cp 113fa0b2aea064b99b98fc7395f0d403f506ae6f2542cc99dc939f8e10375a19:/app/CHECKS /tmp/CHECKS.Xg2D4
remote: + FILENAME=/tmp/CHECKS.Xg2D4/CHECKS
remote: + trap cleanup EXIT
remote: + [[ ! -s /tmp/CHECKS.Xg2D4/CHECKS ]]
remote: + [[ web != \w\e\b ]]
remote: + exec
remote: + read LINE
remote: + [[ / go =~ ^.+= ]]
remote: + read LINE
remote: + ATTEMPT=0
remote: + [[ '' == 1 ]]
remote: + [[ 0 -ge 5 ]]
remote: + FAILEDCHECKS=0
remote: + ATTEMPT=1
remote: + dokku_log_info1 'Attempt 1/5 Waiting for 5 seconds ...'
remote: + echo '-----> Attempt 1/5 Waiting for 5 seconds ...'
remote: + sleep 5
-----> Attempt 1/5 Waiting for 5 seconds ...
remote: + CURL_OPTIONS='-q --compressed --fail --location --max-time 30'
remote: + exec
remote: + read CHECK_URL EXPECTED
remote: + [[ -z / ]]
remote: + [[ / =~ ^# ]]
remote: + [[ / =~ ^(http(s)?:)?/.+ ]]
remote: + continue
remote: + read CHECK_URL EXPECTED
remote: + '[' 0 -gt 0 ']'
remote: + SUCCESS=1
remote: + [[ 1 == 1 ]]
remote: + '[' 0 -gt 0 ']'
remote: + dokku_log_info1 'All checks successful!'
remote: + echo '-----> All checks successful!'
remote: + cleanup
remote: + rm -rf /tmp/CHECKS.Xg2D4
-----> All checks successful!
...
```

SonicHedgehog commented:
@michaelshobbs [The URL is being evaluated against this regular expression](https://github.com/progrium/dokku/blob/v0.3.21/plugins/checks/check-deploy#L134-L136):

```
remote: + [[ / =~ ^(http(s)?:)?/.+ ]]
remote: + continue
```

However, because `/` doesn’t match the `.+` part of the regular expression, it will `continue` to the next URL.

You can clearly see the difference in CircleCI’s logs, [here’s the latest successful build from `master`](https://circleci.com/gh/progrium/dokku/830):

```
+ git push target master
…
remote: -----> Running pre-flight checks
remote: -----> Attempt 1/5 Waiting for 5 seconds ...
remote: -----> All checks successful!
…
remote: =====> Application deployed:
remote:        http://test-go-6532.dokku.me
remote: 
To dokku@dokku.me:test-go-6532
 * [new branch]      master -> master
```

[And the latest build from this pull request](https://circleci.com/gh/progrium/dokku/805):

```
+ git push target master
…
remote: -----> Running pre-flight checks
remote: -----> Attempt 1/5 Waiting for 5 seconds ...
remote:        CHECKS expected result:
remote:        http://localhost/ => "go"
remote: -----> All checks successful!
…
remote: =====> Application deployed:
remote:        http://test-go-32344.dokku.me
remote: 
To dokku@dokku.me:test-go-32344
 * [new branch]      master -> master
```

michaelshobbs commented:
Ah good call. Thanks for pointing that out.
